.travis.yml

```diff
@@ -9,23 +9,30 @@ cache:
 
 matrix:
   include:
+      # strict compilation
     - sudo: false
       install: true
-      script:
-        # The script execution consists of 4 steps:
-        # 1) Increase the jvm max heap size. Strict compilation in 2) requires more than 2 GB.
-        # 2) Strict compilation using error-prone
-        # 3) Reset the maven options. Parallel testing in 4) launches 2 processes, each requires 1 GB of memory. Increasing the maximum memory may cause OutOfMemory error because the instance of Travis has 4 GB of memory space.
-        # 4) Parallel unit testing
-        # Using && instead of independent script steps to make Travis build fail faster, if each step is not successful
-        - echo "MAVEN_OPTS='-Xmx3000m'" > ~/.mavenrc && mvn clean -Pstrict -pl '!benchmarks' compile test-compile -B && rm ~/.mavenrc && mvn test -B -Pparallel-test -Dmaven.fork.count=2
+      # Strict compilation requires more than 2 GB
+      script: echo "MAVEN_OPTS='-Xmx3000m'" > ~/.mavenrc && mvn clean -Pstrict -pl '!benchmarks' compile test-compile -B
+
+      # processing module test
+    - sudo: false
+      install: mvn install -q -ff -DskipTests -B
+      script: mvn test -B -Pparallel-test -Dmaven.fork.count=2 -pl processing
+
+      # non-processing modules test
+    - sudo: false
+      install: mvn install -q -ff -DskipTests -B
+      script: mvn test -B -Pparallel-test -Dmaven.fork.count=2 -pl '!processing'
+
+      # run integration tests
     - sudo: required
       services:
         - docker
       env:
         - DOCKER_IP=172.17.0.1
       install:
         # Only errors will be shown with the -q option. This is to avoid generating too many logs which make travis build failed.
-        - mvn clean install -q -ff -DskipTests -B
+        - mvn install -q -ff -DskipTests -B
```

Comment:
Every build on travis initializes the project directory by cloning druid's git repository, and thus there will be nothing to be cleaned here.